src/Microsoft.AspNetCore.Mvc.Razor/Compilation/IViewCompilerProvider.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
+{
+    public interface IViewCompilerProvider
```

Comment:
I was following the same pattern as the `ICompilerCacheProvider`. Instantiating the cache requires reading precompiled view metadata and we wanted to avoid doing that at startup (because it involves a bunch of disk I/O looking for the precompiled assembly). The alternative was to lazy initialize things inside the `RazorViewCompiler` which ends up being messier.